### PR TITLE
Refactor ChunkServerServiceClient to two separate classes, one for the client side and one for the master side

### PR DESCRIPTION
**Pull request by xicheng87**
In this PR, we split ChunkServerServiceClient into two separate classes, namely ChunkServerServiceMasterClient and ChunkServerServiceClientClient. The former serves as a client wrapper for the master to issue rpc requests to chunk server, and the latter works as a client wrapper for the client to issue rpc requests to chunk server. 

The reason for this splitting is that ChunkServerService includes both requests made by the master (GrantLease, RevokeLease, AdvanceFileChunkVersion, InitFileChunk) and requests made by the client (ReadFile, WriteFile). Having one client-sidie objects for both may not be idea, as a client is not supposed to issue GrantLease request. This refactor serves the purpose of separating the functionalities on the client side of this ChunkServerService.

### CODE REVIEW

File: examples/BUILD.bazel [hunk 1]
```diff
@@ -20,7 +20,8 @@ cc_binary(
     srcs = ["run_mock_protocol_client_main.cc"],
     deps = [
         "//src/common:system_logger",
-        "//src/common/protocol_client:chunk_server_service_client",
+        "//src/common/protocol_client:chunk_server_service_client_client",
```
Comment (Michael-Tu) [inline examples/BUILD.bazel:23]: Can we rename it to something like gfs client

Comment (xicheng87) [reply examples/BUILD.bazel:23]: You meant "chunk_server_service_client_gfs_{client,master,chunk_server}"?

Comment (Michael-Tu) [reply examples/BUILD.bazel:23]: "chunk_server_service_{gfs, master_server}_client

**Comment by Michael-Tu** [comment]
Proposed changes in #34

**Pull request merged**